### ISSUE TITLE
Generate DAO.fetchOneByXYZ(T value) for unique columns on SQLite

### ISSUE THREAD

**Issue by tomas-silhavy** [jOOQ/jOOQ#6632]
### Expected behavior:
Generated UserDao class should also have a method: 
`public UserPojo fetchOneByName(String name);`

### Actual behavior
UserDao has only method:
`List<UserPojo> fetchByName(String ...name);`

### Steps to reproduce the problem:
```
CREATE TABLE user (
  id   INTEGER PRIMARY KEY AUTOINCREMENT,
  name VARCHAR(255) NOT NULL UNIQUE
);
```

let jooq generate Daos.
### Versions:

- jOOQ: 3.10.0 (also not working on 3.9)
- Database - JDBC Driver : sqlite-jdbc 3.19.3

**Comment by lukaseder** [comment]
Thank you very much for your suggestion. That certainly makes sense for single-column unique constraints. I'm not sure if it's reasonable to do this also for composite unique constraints, or even function-based unique constraints, though. Will think about this.

**Comment by tomas-silhavy** [comment]
My point was more about this function (on single unique column) not working on sqlite, because I know it generates it nicely from mysql database.

**Comment by lukaseder** [comment]
Thanks for the update, my bad. You're absolutely right. The problem is that in the past, we hadn't found a way to access SQLite's unique constraints. E.g. the `pragma table_info('table_name')` command doesn't list them.

It appears, though, that more recent SQLite versions now offer a solution for this. E.g. this query should work:

```sql
SELECT *
  FROM sqlite_master AS m,
       pragma_index_list(m.name) AS il,
       pragma_index_info(il.name) AS ii
 WHERE m.type='table'
 AND origin = 'u'
 ORDER BY 1;
```

This is also covered by https://github.com/jOOQ/jOOQ/issues/717.

In fact, once unique keys are available, these DAO methods will be generated as well.

**Comment by lukaseder** [comment]
Yes, I can confirm: Once unique keys are supported for SQLite, these `DAO` methods will be generated as well.

**Comment by lukaseder** [comment]
This will be implemented through #717

**Comment by lukaseder** [comment]
@vptp Yes, please open a new issue
